withTime.js

```diff
@@ -0,0 +1,24 @@
+const EE = require('./eventEmitter');
+
+class WithTime extends EE {
+    async execute(fn) {
+        this.emit('start');
+
+        const startDate = Date.now();
```

Comment:
It would be better to use [perfomance.now](https://nodejs.org/api/perf_hooks.html#performancenow) in execution time measurments

Reply:
done